### PR TITLE
HW flask_4. Add many func, work with sqlalchemy

### PR DESCRIPTION
**Pull request by NykoEugen**
Made 5 points so far

### CODE REVIEW

File: flask_hw/views.py [hunk 1]
```diff
@@ -37,56 +39,80 @@ def login():
 
 def session_username(func):
     @wraps(func)
-    def wrapper():
+    def wrapper(*args, **kwargs):
```
Comment (acman) [inline flask_hw/views.py:42]: 👍

**Review by acman — APPROVED** [review]
GJ!

File: flask_hw/views.py [hunk 2]
```diff
@@ -239,6 +239,72 @@ def params():
...
+        author = request.form['author']
+        price = request.form['price']
+
+    query = Books(title=title, author=author, price=price)
+    db.session.add(query)
+    db.session.commit()
+    return 'Book was add successful'
+
+
+@app.post('/purchases')
```
Comment (acman) [inline flask_hw/views.py:281]: task is about creating purchases. Before creation a purchase, the user and book should be validated.
I think there should be two queries to check the user and book if they are present in DB and return 400 if are not. and then create purchase

**Review by acman — APPROVED** [review]

File: flask_hw/views.py (hunk 2 above)
Comment (NykoEugen) [reply flask_hw/views.py:281]: Ouu, I understand now)

**Pull request merged**